packages/@lwc/engine-core/src/framework/hydration.ts

```diff
@@ -505,7 +505,12 @@ function validateClassAttr(vnode: VBaseElement, elm: Element, renderer: Renderer
     let nodesAreCompatible = true;
     let readableVnodeClassname;
 
-    const elmClassName = getProperty(elm, 'className');
+    const rawElmClassName: string | SVGAnimatedString = getProperty(elm, 'className');
+
+    // In <svg>s, the `class` attribute becomes an SVGAnimatedString
+    const elmClassName =
+        rawElmClassName instanceof SVGAnimatedString ? rawElmClassName.baseVal : rawElmClassName;
```

Comment:
[SVGAnimatedString](https://developer.mozilla.org/en-US/docs/Web/API/SVGAnimatedString) is not supported in IE, but neither is hydration.